x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/rare_job_creator.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SavedSearchSavedObject } from '../../../../../../common/types/kibana';
+import { JobCreator } from './job_creator';
+import { Field, SplitField, Aggregation } from '../../../../../../common/types/fields';
+import { Job, Datafeed, Detector } from '../../../../../../common/types/anomaly_detection_jobs';
+import { JOB_TYPE, CREATED_BY_LABEL } from '../../../../../../common/constants/new_job';
+import { getRichDetectors } from './util/general';
+import { IndexPattern } from '../../../../../../../../../src/plugins/data/public';
+import { isSparseDataJob } from './util/general';
+import { ML_JOB_AGGREGATION } from '../../../../../../common/constants/aggregation_types';
+
+export class RareJobCreator extends JobCreator {
+  private _rareField: Field | null = null;
+  private _populationField: SplitField = null;
+  private _splitField: SplitField = null;
+
+  protected _type: JOB_TYPE = JOB_TYPE.RARE;
+  private _rareInPopulation: boolean = false;
+  private _frequentlyRare: boolean = false;
+  private _rareAgg: Aggregation;
+  private _freqRareAgg: Aggregation;
+
+  constructor(
+    indexPattern: IndexPattern,
+    savedSearch: SavedSearchSavedObject | null,
+    query: object
+  ) {
+    super(indexPattern, savedSearch, query);
+    this.createdBy = CREATED_BY_LABEL.RARE;
+    this._wizardInitialized$.next(true);
+    this._rareAgg = {} as Aggregation;
+    this._freqRareAgg = {} as Aggregation;
+  }
+
+  public setDefaultDetectorProperties(rare: Aggregation | null, freqRare: Aggregation | null) {
+    if (rare === null || freqRare === null) {
+      return;
```

Comment:
Do you still want to return here if one of these is not null?

Reply:
this was copied from the categorisation wizard which does something similar. `null` should never be passed in but the types, it is something has gone wrong with he new job caps endpoint and `rare` and `freq_rare` aggs can't be found.
I think i'll change this function and the one in the categorisation wizard to throw an error here.

Reply:
Changed in https://github.com/elastic/kibana/pull/100390/commits/0862bb9c11cf16d188736577af9995789dd1a95e